app/src/main/java/com/example/atheneum/fragments/BorrowerPageFragment.java

```diff
@@ -58,64 +53,61 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+         * Retrieve request
          */
         FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
         final FirebaseDatabase db = FirebaseDatabase.getInstance();
         DatabaseReference ref = db.getReference().child(getString(R.string.db_requestCollection)).child(currentUser.getUid());
 
         /**
-         * looping the request list
+         * Get the request list
          */
-
         ref.addValueEventListener(new ValueEventListener() {
             @Override
             public void onDataChange(DataSnapshot dataSnapshot) {
                 requestList.clear();
                 for (DataSnapshot item: dataSnapshot.getChildren()) {
                     Request requestItem = new Request();
-                    long leastSigBits = item.child(getString(R.string.db_book_bookID))
-                            .child(getString(R.string.db_book_bookID_leastSigBits))
-                            .getValue() != null ? (long)item.child(
-                            getString(R.string.db_book_bookID))
-                            .child(getString(R.string.db_book_bookID_leastSigBits))
-                            .getValue() : null;
-                    long mostSigBits = item.child(getString(R.string.db_book_bookID))
-                            .child(getString(R.string.db_book_bookID_mostSigBits))
-                            .getValue() != null ? (long)item.child(
-                            getString(R.string.db_book_bookID))
-                            .child(getString(R.string.db_book_bookID_mostSigBits))
-                            .getValue() : null;
-                    UUID requestID = new UUID(mostSigBits, leastSigBits);
-                    requestItem.setBookID(requestID);
-                    User requester = item.child(getString(R.string.db_book_requester)).getValue(User.class);
+//                    long leastSigBits = item.child(getString(R.string.db_book_bookID))
+//                            .child(getString(R.string.db_book_bookID_leastSigBits))
+//                            .getValue() != null ? (long)item.child(
+//                            getString(R.string.db_book_bookID))
+//                            .child(getString(R.string.db_book_bookID_leastSigBits))
+//                            .getValue() : null;
+//                    long mostSigBits = item.child(getString(R.string.db_book_bookID))
+//                            .child(getString(R.string.db_book_bookID_mostSigBits))
+//                            .getValue() != null ? (long)item.child(
+//                            getString(R.string.db_book_bookID))
+//                            .child(getString(R.string.db_book_bookID_mostSigBits))
+//                            .getValue() : null;
+//                    UUID requestID = new UUID(mostSigBits, leastSigBits);
+
+                    String bookID = item.child(getString(R.string.db_book_bookID)).getValue(String.class);
+                    requestItem.setBookID(bookID);
```

Comment:
Why is necessary to manually parse the class like this? Can't you just do Request request = dataSnapshot.getValue(Request.class)?

Reply:
Yeah, i just tried it and you can cast it directly in your loop as 
```
requestList.add(item.getValue(Request.class));
```